Gems/AtomTressFX/Code/Rendering/HairDispatchItem.cpp

```diff
@@ -45,15 +45,15 @@ namespace AZ
                     elementsAmount, 1, 1,
                     TRESSFX_SIM_THREAD_GROUP_SIZE, 1, 1
                 );
-                m_dispatchItem.m_arguments = dispatchArgs;
+                m_dispatchItem.SetArguments(dispatchArgs);
                 RHI::PipelineStateDescriptorForDispatch pipelineDesc;
                 m_shader->GetVariant(RPI::ShaderAsset::RootShaderVariantStableId).ConfigurePipelineState(pipelineDesc);
-                m_dispatchItem.m_pipelineState = m_shader->AcquirePipelineState(pipelineDesc)->GetDevicePipelineState(RHI::MultiDevice::DefaultDeviceIndex).get();
-                m_dispatchItem.m_shaderResourceGroupCount = 2;      // the per pass will be added by each pass.
-                m_dispatchItem.m_shaderResourceGroups = {
-                    hairGenerationSrg->GetRHIShaderResourceGroup()->GetDeviceShaderResourceGroup(RHI::MultiDevice::DefaultDeviceIndex).get(), // Static generation data
-                    hairSimSrg->GetRHIShaderResourceGroup()->GetDeviceShaderResourceGroup(RHI::MultiDevice::DefaultDeviceIndex).get()         // Dynamic data changed between passes
+                m_dispatchItem.SetPipelineState(m_shader->AcquirePipelineState(pipelineDesc));
+                AZStd::array<const RHI::MultiDeviceShaderResourceGroup*, 2> srgs{
+                    hairGenerationSrg->GetRHIShaderResourceGroup(), // Static generation data
+                    hairSimSrg->GetRHIShaderResourceGroup() // Dynamic data changed between passes
                 };
+                m_dispatchItem.SetShaderResourceGroups(srgs, 2);
```

Comment:
Why do we need to pass the number of srgs? Can't we use the size of the span that the `SetShaderResourceGroups` function gets?

Reply:
Good point, fixed it!